### PR TITLE
Drawbox

### PR DESCRIPTION
**Pull request by wnordmann**
Adding in ability to draw a box to meet needed for mobile

### CODE REVIEW

File: __tests__/components/map-drawing.test.js [hunk 1]
```diff
@@ -129,6 +129,50 @@ if (HAS_CANVAS) {
...
+
+      // if the drawing control has been added to the map then
+      //  there should be 1 additional interaction.
+      expect(ol_map.getInteractions().getLength()).toBe(n_interactions + 1);
+
+      // get the last interaction, which should be the drawing interaction
+      const draw = ol_map.getInteractions().item(n_interactions);
+
+      const dummy_feature = new Feature();
+      dummy_feature.setGeometry(new Polygon(5, 5));
```
Comment (bartvde) [inline __tests__/components/map-drawing.test.js]: this is not a valid Polygon, argument should at least be an array

File: __tests__/components/map-drawing.test.js [hunk 2]
```diff
@@ -129,6 +129,50 @@ if (HAS_CANVAS) {
...
+      draw.dispatchEvent({
+        type: 'drawend',
+        feature: dummy_feature,
+      });
+
+      store.dispatch(MapActions.setView([-45, -45], 11));
```
Comment (bartvde) [inline __tests__/components/map-drawing.test.js]: what is tested here actually? I don't see any expect anymore at the end.

File: src/components/map.js [hunk 3]
```diff
@@ -1199,9 +1199,15 @@ export class Map extends React.Component {
 
       this.activeInteractions = [select];
     } else if (INTERACTIONS.drawing.includes(drawingProps.interaction)) {
-      const draw = new DrawInteraction({
-        type: drawingProps.interaction,
-      });
+      var drawObj = {type: drawingProps.interaction};
+      if (drawingProps.interaction === 'Box'){
```
Comment (bartvde) [inline src/components/map.js]: should use the constant here

File: src/components/map.js [hunk 4]
```diff
@@ -1199,9 +1199,15 @@ export class Map extends React.Component {
 
       this.activeInteractions = [select];
     } else if (INTERACTIONS.drawing.includes(drawingProps.interaction)) {
-      const draw = new DrawInteraction({
-        type: drawingProps.interaction,
-      });
+      var drawObj = {type: drawingProps.interaction};
```
Comment (bartvde) [inline src/components/map.js]: I'd put this in the else statement (also let's use ```let```)

File: src/components/map.js [hunk 5]
```diff
@@ -1199,9 +1199,15 @@ export class Map extends React.Component {
...
+      if (drawingProps.interaction === 'Box'){
+        const geometryFunction = DrawInteraction.createBox();
+        drawObj = {source: drawingProps.sourcename,
```
Comment (bartvde) [inline src/components/map.js]: source should not be set here, it's handled by the ```drawend``` event instead

File: src/components/map.js [hunk 6]
```diff
@@ -1199,9 +1199,15 @@ export class Map extends React.Component {
...
+          type: 'Circle',
+          geometryFunction: geometryFunction
```
Comment (bartvde) [inline src/components/map.js]: can be simply ```geometryFunction,```

File: src/components/map.js (hunk 3 above)
Comment (wnordmann) [reply src/components/map.js]: Totally, at first I thought you meant a const and I was confused

File: src/components/map.js (hunk 4 above)
Comment (wnordmann) [reply src/components/map.js]: Easy enough to put in an else, but it's a really small else.

**Pull request merged**